### PR TITLE
[ENGG-1625] feat:  MV3 - stitch sessions across navigation

### PR DESCRIPTION
**Pull request by nafees87n**
Closes issue: 

## 📜 Summary of changes:
- Implemented tabService
- Sessions are stitched across navigation

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: browser-extension/mv3/src/service-worker/services/tabService.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    const tab = this.getTab(tabId);
+
+    if (!tab || !tab[DataScope.TAB]) {
+      return;
+    }
+
+    delete tab[DataScope.TAB][key];
+  }
+}
+const tabService = TabService.getInstance();
```
Comment (lazyvab) [inline browser-extension/mv3/src/service-worker/services/tabService.ts:192]: Each ES module is executed once, so already a singleton. No need to write `getInstance` logic.

```js
export const tabService = new TabService();
```

File: browser-extension/mv3/src/service-worker/services/sessionRecording.ts [hunk 2]
```diff
@@ -78,32 +73,102 @@ export const watchRecording = (tabId: number) => {
...
+  if (!sessionRecordingData) {
+    const sessionRecordingConfig = await getSessionRecordingConfig(tab.url);
+
+    if (sessionRecordingConfig) {
+      sessionRecordingData = { config: sessionRecordingConfig, url: tab.url };
+      const recordingMode = sessionRecordingConfig?.autoRecording?.mode;
+
+      sessionRecordingData.showWidget = recordingMode === "custom";
+
+      if (recordingMode === "allPages") {
```
Comment (lazyvab) [inline browser-extension/mv3/src/service-worker/services/sessionRecording.ts:130]: any enum type defined for recordingMode config?

**Review by lazyvab — APPROVED** [review]
Looks mostly in sync with MV2 code. Left minor code styling suggestions.

File: browser-extension/mv3/src/service-worker/services/sessionRecording.ts (hunk 2 above)
Comment (nafees87n) [reply browser-extension/mv3/src/service-worker/services/sessionRecording.ts:130]: yes we have

**Pull request merged**